asgi_redis/sentinel.py

```diff
@@ -0,0 +1,142 @@
+    """
+    Variant of the Redis channel layer that supports the Redis Sentinel HA protocol.
+
+    Supports sharding, but assumes that there is only one sentinel cluster with multiple redis services
+    monitored by that cluster. So, this will only connect to a single cluster of Sentinel servers,
+    but will suppport sharding by asking that sentinel cluster for different services. Also, any redis connection
+    options (socket timeout, socket keepalive, etc) will be assumed to be identical across redis server, and
+    across all services.
+
+    "hosts" in this arrangement, is used to list the redis sentinel hosts. As such, it only supports the
+    tuple method of specifying hosts, as that's all the redis sentinel python library supports at the moment.
+
+    "services" is the list of redis services monitored by the sentinel system that redis keys will be distributed
+     across.
+    """
+    def __init__(
+                self,
+                expiry=60,
+                hosts=None,
+                prefix="asgi:",
+                group_expiry=86400,
+                capacity=100,
+                channel_capacity=None,
+                symmetric_encryption_keys=None,
+                stats_prefix="asgi-meta:",
+                socket_connect_timeout=None,
+                socket_timeout=None,
+                socket_keepalive=None,
+                socket_keepalive_options=None,
+                services=None):
+            self.services = self._setup_services(services)
+
+            # Precalculate some values for ring selection
+            self.ring_size = len(self.services)
+
+            self.hosts = self._setup_hosts(hosts)
+
+            self._sentinel = self._generate_sentinel(
+                redis_kwargs={
+                    "socket_connect_timeout": socket_connect_timeout,
+                    "socket_timeout": socket_timeout,
+                    "socket_keepalive": socket_keepalive,
+                    "socket_keepalive_options": socket_keepalive_options,
+                },
+            )
+
+            super(RedisSentinelChannelLayer, self).__init__(expiry,
+                                                            hosts,
+                                                            prefix,
+                                                            group_expiry,
+                                                            capacity,
+                                                            channel_capacity,
+                                                            symmetric_encryption_keys,
+                                                            stats_prefix,
+                                                            socket_connect_timeout,
+                                                            socket_timeout,
+                                                            socket_keepalive,
+                                                            socket_keepalive_options)
```

Comment:
Could you indent these to match the existing code style (that is, just bring them in 4 spaces compared to the outer scope, rather than aligning them all with the opening parenthesis?)